client/fs_endpoint.go

```diff
@@ -177,11 +177,20 @@ func (f *FileSystem) stream(conn io.ReadWriteCloser) {
 		handleStreamResultError(allocIDNotPresentErr, pointer.Of(int64(400)), encoder)
 		return
 	}
-	alloc, err := f.c.GetAlloc(req.AllocID)
+
+	ar, err := f.c.getAllocRunner(req.AllocID)
 	if err != nil {
 		handleStreamResultError(structs.NewErrUnknownAllocation(req.AllocID), pointer.Of(int64(404)), encoder)
 		return
 	}
+	if ar.IsDestroyed() {
+		handleStreamResultError(
+			fmt.Errorf("allocation %s destroyed from client", req.AllocID),
```

Comment:
Not sure if this is the best message to return. I think GC is only one of the reasons the alloc runner may have been destroyed.

Reply:
I wonder if something along the lines of "allocation %s not found on client" would be better? Destroyed seems like an internal aspect of the client which operators might find confusing to the idea that an alloc is not found.

Reply:
Yeah, I wasn't happy with the message. You suggestion is good, I just adjusted it to `state for allocation %s not found on client` because, weirdly, the alloc is still there, it's just that the GC deletes the state (alloc runner)